src/runtime_src/core/tools/common/ReportAieShim.cpp

```diff
@@ -99,6 +99,13 @@ ReportAieShim::writeReport( const xrt_core::device* /*_pDevice*/,
       _output << fmt4("%d") % "Row" % tile.second.get<int>("row");
       if(tile.second.find("dma") != tile.second.not_found()) {
         _output << boost::format("    %s:\n") % "DMA";
+
+	_output << boost::format("        %s:\n") % "FIFO";
+
+	auto& fifo_node  = tile.second.get_child("dma.fifo_info");
+	_output << fmt16("%s") % "Counter0" % fifo_node.get<std::string>("fifo_counter0");

```

Comment:
This code should be refactored to that it can report 1, 3 or more counters. Yes, I understand that this currently the case, but there is too much hard-coding here and it needs to be more dynamic.

Reply:
Same as above

Reply:
ditto ;)

Reply:
Fixed. Thanks